docs/tutorials/rest_api/index.md

```diff
@@ -0,0 +1,405 @@
+perform several interesting queries on them, and read from the incremental views.
+
+We'll be going through the following steps:
+
+1. Create and compile an SQL program
+2. Create data connectors
+3. Create a pipeline with the prepared program and connectors
+4. Start the pipeline
+5. Check pipeline progress
+6. Read data directly from an incremental view
+7. Feed data directly into a stream table
+8. Cleanup
+
+... all using just `curl`!
+
+> Note: at any point in the tutorial, don't forget you can check out the
+Web Console by visiting `http://localhost:8080` in your browser!
+
+### Step 1: SQL program
+
+The SQL program defines the input stream tables using `CREATE TABLE` statements
+and incremental views using `CREATE VIEW` statements. We will create three tables,
+namely `vendor`, `part`, and `price`. As views, we'll create `low_price` which lists
+the lowest available price for each part across all vendors, and `preferred_vendor` which
+supplements the lowest price by adding vendor information.
+
+We create the program named `sc-program` as follows:
+
+```
+curl -i -X PUT http://localhost:8080/v0/programs/sc-program \
+-H 'Content-Type: application/json' \
+-d "{
+    \"description\": \"Supply Chain program\",
+    \"code\": \" \
+-- Stream tables -- \n\
+\n\
+CREATE TABLE vendor ( \n\
+    id BIGINT NOT NULL PRIMARY KEY, \n\
+    name VARCHAR, \n\
+    address VARCHAR \n\
+); \n\
+\n\
+CREATE TABLE part ( \n\
+    id bigint NOT NULL PRIMARY KEY, \n\
+    name VARCHAR \n\
+); \n\
+\n\
+CREATE TABLE price ( \n\
+    part BIGINT NOT NULL, \n\
+    vendor BIGINT NOT NULL, \n\
+    price DECIMAL \n\
+); \n\
+\n\
+-- Incremental views -- \n\
+\n\
+CREATE VIEW low_price (part, price) AS \n\
+SELECT \n\
+   price.part AS part, \n\
+   MIN(price.price) AS price \n\
+FROM price \n\
+GROUP BY price.part; \n\
+\n\
+CREATE VIEW preferred_vendor (part_id, part_name, vendor_id, vendor_name, price) AS \n\
+SELECT \n\
+    part.id AS part_id, \n\
+    part.name AS part_name, \n\
+    vendor.id AS vendor_id, \n\
+    vendor.name AS vendor_name, \n\
+    price.price AS price \n\
+FROM \n\
+    price, part, vendor, low_price \n\
+WHERE \n\
+   price.price = low_price.price AND \n\
+   price.part = low_price.part AND \n\
+   part.id = price.part AND \n\
+   vendor.id = price.vendor;\"
+}"
+```
+
+(The many escape characters in the command are to both have the SQL readable enough
+and be correct JSON simultaneously.)
+
+As response, we should get back `HTTP/1.1 201 Created` along with
+the identifier of the program and its version (1).
+When an SQL program is created or when its code is updated, its version is incremented
+and compilation is automatically triggered.
+
+Check that the program is compiled:
+
+```
+curl -X GET http://localhost:8080/v0/programs/sc-program
+```
+
+... which will have `status` set to `Success` when compilation is done.
+
+### Step 2: Data connectors
+
+Data connectors are used to both input data sources into stream tables,
```

Comment:
I don't think we use the term "stream tables". Let's refer to them as just "tables".